utilsd/config/registry.py

```diff
@@ -147,22 +147,63 @@ def dataclass_from_class(cls):
     fields = [
         ('_type', ClassVar[Type], cls),
     ]
+    non_default_fields = []
+    default_fields = []
     init_signature = inspect.signature(cls.__init__)
+    # Track presented param names. The same name may appear in different classes when **kwargs is passed.
+    existing_names = dict()
+    expand_super = False
```

Comment:
Suggest adding a parameter to `dataclass_from_class`  that can optionally disable this auto-expand behavior.